### PR TITLE
Task 02: Area of a triangle

### PR DESCRIPTION
**Pull request by RakValery**
Task02v1.0

### CODE REVIEW

File: tasks/task02/Area_of_a_triangle.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# main menu
+import math
+print("Welcome to the triangle area calculation tool.")
+while True:
+    print("\n","Menu:","1. Calculate triangle area by base and height","2. Calculate triangle area by 2 sides and angle between them","3. Exit", sep = "\n")
+    print("Enter menu item number: ", end = "")
+    mitem = input()
```
Comment (minotru) [inline tasks/task02/Area_of_a_triangle.py]: `input` has `prompt` parameter so that you can write:

`mitem = input("Enter menu item number:")`

Comment (minotru) [reply tasks/task02/Area_of_a_triangle.py]: https://www.w3schools.com/python/ref_func_input.asp

File: tasks/task02/Area_of_a_triangle.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if mitem == "1":
+        print("Enter base and height (space separated): ", end = "")
+        while True:
+            inp = input().split(" ")
+            try:
+                base, height = [float(x) for x in inp]
```
Comment (minotru) [inline tasks/task02/Area_of_a_triangle.py]: here you use list comprehension `[f(x) for x in seq]` to iterate and convert to float. it is fine, but `list` will allocate extra memory that is not needed here, the list is not reused anywhere.

for such cases [generator expression](https://www.python.org/dev/peps/pep-0289/) is a better choice. generators just generate a sequence of values on the fly, it is not stored in memory. more about generators vs lists: https://stackoverflow.com/questions/47789/generator-expressions-vs-list-comprehensions

that is, in this case it is better to write `base, height = (float(x) for x in inp)`.

bonus: there is a built-in function `map` that applies a function to every element of the sequence and produces a generator. so another way to write is:
```python
base, height = map(float, inp) # an equivalent of (float(x) for x in inp)
```

https://www.geeksforgeeks.org/python-map-function/

File: tasks/task02/Area_of_a_triangle.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                print("\nbase =", base, " height =", height) 
+                print("Area is: ", base * height / 2)
```
Comment (minotru) [inline tasks/task02/Area_of_a_triangle.py]: an alternative to passing several arguments to `print` is string formatting via f-strings: https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

for example:
```python
print(f"\nbase = {base}, height = {height}")
```

File: tasks/task02/Area_of_a_triangle.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                    err = len(inp)
+                print("\nYou entered ", err," value(s). Please enter 2 numeric values space separated: ", end = "")
+    elif mitem == "2":
+        print("Enter 2 sides and angle(degrees) between them (space separated): ", end = "")
+        while True:
+            inp = input().split(" ")
+            try:
+                side1, side2, angle = [float(x) for x in inp]
+                print("side1 =", side1, " side2 =", side2, " angle =", angle)
+                print("Area is: ", (side1 * side2 * math.sin(angle * math.pi / 180) / 2))
```
Comment (minotru) [inline tasks/task02/Area_of_a_triangle.py]: there is a convenient function to convert degrees to radians: `math.radians`

File: tasks/task02/Area_of_a_triangle.py [hunk 5]
```diff
@@ -1,34 +1,39 @@
 # main menu
 import math
-print("Welcome to the triangle area calculation tool.")
+print("\nWelcome to the triangle area calculation tool.")
```
Comment (minotru) [inline tasks/task02/Area_of_a_triangle.py:3]: by the way, why do you add `\n` in the beginning?

**Review by minotru — APPROVED** [review]
great solution! approved

File: tasks/task02/Area_of_a_triangle.py (hunk 5 above)
Comment (RakValery) [reply tasks/task02/Area_of_a_triangle.py:3]: for beauty

**Pull request merged**